backends/tfhe-cuda-backend/cuda/src/integer/cmux.cuh

```diff
@@ -51,31 +51,35 @@ __host__ void host_integer_radix_cmux_kb(
 
   // Since our CPU threads will be working on different streams we shall assert
   // the work in the main stream is completed
-  cuda_synchronize_stream(streams[0], gpu_indexes[0]);
-  auto true_stream = mem_ptr->zero_if_true_buffer->local_stream;
-  auto false_stream = mem_ptr->zero_if_false_buffer->local_stream;
+  auto true_streams = mem_ptr->zero_if_true_buffer->local_streams;
+  auto false_streams = mem_ptr->zero_if_false_buffer->local_streams;
+  for (uint j = 0; j < gpu_count; j++) {
+    cuda_synchronize_stream(streams[j], gpu_indexes[j]);
```

Comment:
Why do we need to synchronize any device other than 0?

Reply:
We're going to be executing on different streams on each GPU so I bellieve we have to first synchronize the "main" stream on each GPU as well. It's the same logic we had before, except across multiple GPUs.